src/huggingface_hub/HfRepostiory.py

```diff
@@ -0,0 +1,138 @@
+import os
+import subprocess
+from typing import Dict, List
+
+
+MODEL_CARD_TEMPLATE = """
+---
+tags:
+- sagemaker
+datasets:
+- {dataset}
+---
+## {model_id} Trained from SageMaker HuggingFace extension.
+
+#### Hyperparameters
+```json
+{hyperparameters}
+```
+
+#### Eval
+| key | value |
+| --- | ----- |
+{eval_table}
+"""
+
+
+class HfRepository:
+    """Git-based system for HuggingFace Hub repositories"""
+
+    def __init__(
+        self,
+        repo_url: str,
+        huggingface_token: str,
+        model_dir: str,
+    ):
+        self.repo_url = self.add_token_to_repository_url(repo_url, huggingface_token)
+
+        self.model_dir = model_dir
+
+    def init_new_repository(
+        self,
+        large_file_tracking_list=[],
+        user="sagemaker",
+        email="sagemaker@huggingface.co",
+    ):
+        """Initializes a new hf hub repository"""
+
+        self.email = email
+        self.user = user
+        self.large_file_tracking_list = [
+            "*.bin.*",
+            "*.lfs.*",
+            "*.bin",
+            "*.h5",
+            "*.tflite",
+            "*.tar.gz",
+            "*.ot",
+            "*.onnx",
+            "*.pt",
```

Comment:
I'm not sure you need to specify this list here, given it's already added to `.gitattributes` when creating the repo on the hub (see https://github.com/huggingface/moon-landing/blob/master/server/hosted_git/RepoBuilder.ts#L15 for the current list, will be extended by https://github.com/huggingface/moon-landing/issues/273 - will handle this issue today or later this week)